### PR TITLE
feat: Add redirect to password recover page for login with legacy password which does not match new requirements

### PR DESCRIPTION
**Pull request by Zwaen91**
### 1. Why is this change necessary?
Customers with a legacy password that e.g. is not 8 characters long cannot login.

### 2. What does this change do, exactly?
Redirect the customer to the recover page and show a message that the password has to be reset.

### 3. Describe each step to reproduce the issue or behaviour.
Login with customer with legacy password which is lower than 8 characters.

### 4. Please link to the relevant issues (if any).
Issue https://github.com/shopware/platform/issues/2987

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [x] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/workflow/2020-08-03-implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Storefront/Controller/AuthController.php [hunk 1]
```diff
@@ -177,6 +178,10 @@ public function login(Request $request, RequestDataBag $data, SalesChannelContex
             if ($e instanceof CustomerAuthThrottledException) {
                 $waitTime = $e->getWaitTime();
             }
+        } catch (PasswordPoliciesUpdatedException $e) {
+            $this->addFlash(self::WARNING, $this->trans('account.passwordPoliciesUpdated'));
+
+            return $this->forwardToRoute('frontend.account.recover.page');
         }
 
         $data->set('password', null);
```
Comment (aragon999) [inline src/Storefront/Controller/AuthController.php]: Maybe put that unset of the password into the `finally` block of the `try-catch` construct, as in that case now the request is forwarded with the password still in memory and in the data bag.

Comment (Zwaen91) [reply src/Storefront/Controller/AuthController.php]: @aragon999 Thanks for your comment. You're right. I have updated it.

File: src/Storefront/Resources/snippet/en_GB/storefront.en-GB.json [hunk 2]
```diff
@@ -181,6 +181,7 @@
     "emailChangeNoSuccess": "Email address could not be changed.",
     "passwordChangeSuccess": "Your password has been updated.",
     "passwordChangeNoSuccess": "Password could not be changed.",
+    "passwordPoliciesUpdated": "The password policies has changed. Please request a new one.",
```
Comment (MelvinAchterhuis) [inline src/Storefront/Resources/snippet/en_GB/storefront.en-GB.json]: `policies have changed` or `policy has changed`.

Comment (Zwaen91) [reply src/Storefront/Resources/snippet/en_GB/storefront.en-GB.json]: Fixed. Thanks @MelvinAchterhuis

**Comment by keulinho** [comment]
@sydinh please take a look here :pray:

**Comment by PheysX** [comment]
Can we merge this pr in 6.5.0.0? @keulinho

**Comment by mstegmeyer** [comment]
Hey @Zwaen91 and @PheysX , we are truly sorry for dropping the ball here. This PR got lost in one of our teams and we have now reimported it (and rebased it by 3242 commits) and will try to merge it in the following days!

Have a nice weekend ☀️

**Comment by mstegmeyer** [comment]
Hey, your PR has finally been merged. Again my apologies for ignoring it for way too long. Thank you for your contribution 💙 🎉

**Pull request closed without merging**